pkg/yum/repository.go

```diff
@@ -437,7 +439,7 @@ func ParseCompsXML(body io.ReadCloser, url *string) (Comps, error) {
 	fileExtension := filepath.Ext(*url)
 
 	// handle uncompressed comps
-	if fileExtension == ".xml" {
+	if fileExtension == ".xml" || fileExtension == "" {
```

Comment:
It seems like there may not be a hard rule about what the comps file is named. What do you think about passing through the data.type ("group" or "group_gz") and using that for this conditional? I think that would future-proof this for any other weird comps file names.

Reply:
Another option would be to try and detect the file type. We do that for the compressed files here, but that library doesn't seem to support XML.

Reply:
that makes sense, though i see some repos have compressed comps in the "group" data type (https://fixtures.pulpproject.org/rpm-unsigned/repodata/repomd.xml), so i don't think i can rely on that always being consistent :/

Reply:
yea do you mean the filetype library? i tried using that before to handle compressed comps and realized it didn't support xml, which is why i resorted to checking the extension here. i can see if i can do something with that though, similar to ParseCompressedData if it's not gz / zstd / xz then assume it's uncompressed?

Reply:
yeah, good idea! let's try that